plugins/woocommerce/tests/php/includes/gateways/paypal/class-wc-gateway-paypal-test.php

```diff
@@ -223,4 +223,232 @@ public function test_correct_settings_is_displayed_when_orders_v2_is_disabled()
+		$order->save();
+
+		$order_not_paypal = WC_Helper_Order::create_order();
+		$order_not_paypal->set_payment_method( 'bacs' );
+		$order_not_paypal->save();
+
+		$order_missing_paypal_id = WC_Helper_Order::create_order();
+		$order_missing_paypal_id->set_payment_method( WC_Gateway_Paypal::ID );
+		$order_missing_paypal_id->save();
+
+		return array(
+			'order not found'         => array(
+				'order ID'             => 0,
+				'should use orders v2' => true,
+				'mock Jetpack params'  => true,
+				'expect to save'       => false,
+			),
+			'invalid payment method'  => array(
+				'order ID'             => $order_not_paypal->get_id(),
+				'should use orders v2' => true,
+				'mock Jetpack params'  => true,
+				'expect to save'       => false,
+			),
+			'orders v2 not enabled'   => array(
+				'order ID'             => $order->get_id(),
+				'should use orders v2' => false,
+				'mock Jetpack params'  => true,
+				'expect to save'       => false,
+			),
+			'missing PayPal order ID' => array(
+				'order ID'             => $order_missing_paypal_id->get_id(),
+				'should use orders v2' => true,
+				'mock Jetpack params'  => true,
+				'expect to save'       => false,
+			),
+			'exception thrown'        => array(
+				'order ID'             => $order->get_id(),
+				'should use orders v2' => true,
+				'mock Jetpack params'  => false,
+				'expect to save'       => false,
+			),
+			'successful update'       => array(
+				'order ID'             => $order->get_id(),
+				'should use orders v2' => true,
+				'mock Jetpack params'  => true,
+				'expect to save'       => true,
+			),
+		);
+	}
+
+	/**
+	 * Tests for the `enqueue_scripts` method.
+	 *
+	 * @param bool   $gateway_enabled Whether the gateway is enabled.
+	 * @param string $client_id       The client ID to set in options.
+	 * @param bool   $script_expected Whether we expect the script to be enqueued.
+	 * @return void
+	 *
+	 * @dataProvider provide_test_enqueue_scripts
+	 */
+	public function test_enqueue_scripts( $gateway_enabled, $client_id, $script_expected ) {
+		// Enable the gateway.
+		update_option( 'woocommerce_paypal_settings', array( 'enabled' => $gateway_enabled ? 'yes' : 'no' ) );
+
+		// Set cached client ID.
+		update_option( 'woocommerce_paypal_client_id_live', $client_id );
+
+		add_filter( 'woocommerce_is_cart', '__return_true' );
+
+		/**
+		 * @var WC_Gateway_Paypal $mock_gateway Mocked gateway with Orders v2 enabled.
+		 */
+		$mock_gateway = $this->getMockBuilder( WC_Gateway_Paypal::class )
+			->onlyMethods( array( 'should_use_orders_v2' ) )
+			->getMock();
+		$mock_gateway->method( 'should_use_orders_v2' )->willReturn( true );
+		$mock_gateway->testmode = false;
+
+		$mock_gateway->enqueue_scripts();
+
+		// Clean up.
+		delete_option( 'woocommerce_paypal_settings', array() );
+		delete_option( 'woocommerce_paypal_client_id_live' );
+		remove_all_filters( 'woocommerce_is_cart' );
```

Comment:
Should we not remove the specific filter we added? This feels too broad.
```suggestion
		remove_filter( 'woocommerce_is_cart', '__return_true' );
```